homeassistant/components/luxor/light.py

```diff
@@ -0,0 +1,199 @@
+
+    async def async_update_group_data():
+        """Fetch group data from Luxor controller."""
+        groups = await async_safe_fetch(client.get_groups)
+        return {group["Grp"]: group for group in groups}
+
+    light_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="luxor",
+        update_method=async_update_group_data,
+        update_interval=timedelta(seconds=1),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    #
+    # If the refresh fails, async_config_entry_first_refresh will
+    # raise ConfigEntryNotReady and setup will try again later
+    await light_coordinator.async_config_entry_first_refresh()
+
+    entities = [
+        LuxorGroupLightEntity(light_coordinator, client, group_id)
+        for group_id in light_coordinator.data.keys()
+    ]
+    async_add_entities(entities)
+
+    #
+    # Add Luxor themes as lights, if configured to do so.
+    # Users might want avoid adding them if they do things like 'turn on all lights'
+    # that will result in enabling all the themes in parellel, which is not ideal.
+    #
+    async def async_update_theme_data():
+        """Fetch theme data from Luxor controller."""
+        themes = await async_safe_fetch(client.get_themes)
+        return {theme["ThemeIndex"]: theme for theme in themes}
+
+    include_luxor_themes = config_entry.options.get(
+        CONF_INCLUDE_LUXOR_THEMES, DEFAULT_INCLUDE_LUXOR_THEMES
+    )
+
+    if include_luxor_themes:
+        theme_coordinator = DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            name="luxor",
+            update_method=async_update_theme_data,
+            update_interval=timedelta(seconds=1),
+        )
+
+        await theme_coordinator.async_config_entry_first_refresh()
+
+        entities = [
+            LuxorThemeLightEntity(theme_coordinator, client, theme_id)
+            for theme_id in theme_coordinator.data.keys()
+        ]
+        async_add_entities(entities)
+
+    return True
+
+
+async def async_safe_fetch(method):
+    """Safely fetch data."""
+    try:
+        # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+        # handled by the data update coordinator
+        async with async_timeout.timeout(10):
+            return await method()
+    except aio_exceptions.ClientConnectorError as err:
+        raise UpdateFailed(f"Connection error: {err}")
+    except LuxorError as err:
+        raise UpdateFailed(f"Error communicating with API: {err}")
+
+
+def luxor_brightness_to_hass(value):
+    """Convert luxor brightness 1..100 to hass format 0..255."""
+    return int((value * 255) // 100)
+
+
+def hass_to_luxor_brightness(value):
+    """Convert hass brightness 0..255 to luxor 1..100 scale."""
+    return int(round((value * 100) / 255))
+
+
+class LuxorGroupLightEntity(CoordinatorEntity, LightEntity):
+    """Representation of a Luxor light group as a homeassisant light entity."""
+
```

Comment:
```suggestion
class LuxorGroupLightEntity(CoordinatorEntity, LightEntity):
    """Representation of a Luxor light group as a homeassisant light entity."""

    _attr_supported_features = SUPPORT_BRIGHTNESS
```

However!

We are moving away from supported features for lights. Please use color modes instead (but can be set in a similar fashion as shown in this code comment).